src/com/felspar/android/Setting.java

```diff
@@ -10,5 +10,15 @@
 
 public class Setting {
     public native static boolean fromJSON(String domain, String json);
+
     public native static boolean fromString(String domain, String section, String name, String value);
+
+    private native static String readString(String section, String name);
+
+    public static String readStringFrom(String section, String name) throws Exception {
+        String value = readString(section, name);
+        if(!value.startsWith("Bearer "))
```

Comment:
This check needs to be in your application code. We are likely to have all sorts of settings whose values don't start with "Bearer"